src/views/_shared/DynamicScrollToTop.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { useEffect } from "react";
+import { withRouter } from "react-router-dom";
+
+//This component scrolls to the top of the page when a link is clicked
+//unless it is a hash link
+const DynamicScrollToTop = (props: any) => {
```

Comment:
Destructure `props` into `location` and `history`.
`const DynamicScrollToTop = ({location, history}) => {`

Doing this will make other parts of the code less verbose.